library/common/network/configurator.cc

```diff
@@ -181,25 +181,32 @@ void Configurator::reportNetworkUsage(envoy_netconf_t configuration_key, bool ne
 }
 
 void Configurator::onDnsResolutionComplete(
-    const std::string& host, const Extensions::Common::DynamicForwardProxy::DnsHostInfoSharedPtr&,
+    const std::string& resolved_host,
+    const Extensions::Common::DynamicForwardProxy::DnsHostInfoSharedPtr&,
     Network::DnsResolver::ResolutionStatus) {
-  if (enable_drain_post_dns_refresh_ && pending_drain_) {
-    pending_drain_ = false;
+  if (enable_drain_post_dns_refresh_) {
+    // Check if the set of hosts pending drain contains the current resolved host.
+    if (hosts_to_drain_.erase(resolved_host) == 0) {
```

Comment:
You're correct that this would be wrong if this was a vector, but `hosts_to_drain_` is an `absl::flat_hash_set`.

Here is what is being called:
https://github.com/abseil/abseil-cpp/blob/master/absl/container/flat_hash_set.h#L232

Reply:
Another way to read this is "return if nothing was erased".

Reply:
ok thanks for sharing that. I still haven't come around to appreciating this overall C++ pattern of returning iterators from functions, but this isn't the place to debate that.

Reply:
I agree working with iterators is a bit of a mental adjustment. That said, in this case, the function is simply returning the number of elements that were deleted as a `size_type`. :)